### PR TITLE
Added a slack function

### PR DESCRIPTION
**Pull request by LindsayYoung**
This slack function will slacks developers about the success or failure of the nightly updates. It should be easy to integrate for other things where we want more visibility .

### CODE REVIEW

**Comment by LindsayYoung** [comment]
needs web hook to be added to env vars

**Comment by lbeaufort** [comment]
I added the `SLACK_HOOK` credentials to `dev`, `stage`, and `prod`.

**Comment by LindsayYoung** [comment]
Mandril is not shut off if we want to add some of that back

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ development tools:
 We currently have several monitoring tools in place:
 
 - New Relic, mainly for basic pings
-- A custom script powered by Mandrill to e-mail us nightly status reports in
+- A custom script powered by slacks us nightly status reports about
```
Comment (ccostino) [inline CONTRIBUTING.md]: Very nice to see the whitespace cleanup in here!

For this particular line, it looks like we're missing a word or two.  What about this to make it easier to understand?

`A custom script powered by Slack that posts a nightly status update about...`

File: manage.py [hunk 2]
```diff
@@ -407,5 +408,13 @@ def load_efile_sheets():
         df.to_json(path_or_buf="data/" + table + ".json", orient='values')
 
 
+@manager.command
+def slack_message(message):
+    """ Sends a message to the bots channel. you can add this command to ping you when a task is done, etc.
+    run ./manage.py slack_message 'The message you want to post'
+    """
+    post_to_slack(message, '#bots')
```
Comment (ccostino) [inline manage.py]: Oh awesome, this is a great idea!  Also, should we set the default channel as an environment variable instead?

File: requirements.txt [hunk 3]
```diff
@@ -12,13 +12,11 @@ sqlalchemy-postgres-copy==0.3.0
 networkx==1.11
 newrelic==2.96.0.80
 SQLAlchemy==1.0.17
-mandrill==1.0.57
 icalendar==3.9.1
 GitPython==1.0.1
 gunicorn==19.7.1
 gevent==1.2.2
 webargs==0.18.0
-slacker==0.8.6
```
Comment (ccostino) [inline requirements.txt]: Just confirming that we don't need this anymore as we're just posting directly over HTTPS to Slack now?

File: webservices/common/views.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import sys
```
Comment (ccostino) [inline webservices/common/views.py]: Looks like an extraneous `import` statement that's not used made it in here.

File: webservices/config.py [hunk 5]
```diff
@@ -32,7 +32,6 @@ def get_cycle_end(year):
 REQUIRED_CREDS = (
     'SQLA_CONN',
     'FEC_SLACK_TOKEN',
-    'MANDRILL_API_KEY',
```
Comment (ccostino) [inline webservices/config.py]: Do we need to add the `SLACK_HOOK` environment variable here in addition to the `FEC_SLACK_TOKEN` that's already here, or no?

File: webservices/tasks/refresh.py [hunk 6]
```diff
@@ -20,9 +22,10 @@ def refresh():
             manage.refresh_itemized()
             manage.refresh_materialized()
             download.clear_bucket()
+            slack_message = '*Success* nightly updates for {0} completed'.format(env.get_credential(NEW_RELIC_APP_NAME))
+            utils.post_to_slack(slack_message, '#bots')
```
Comment (ccostino) [inline webservices/tasks/refresh.py]: Just confirming that `NEW_RELIC_APP_NAME` is being used to identify the app here.  Also, as noted above, should we set a default environment variable for the channel name to post to?

File: webservices/tasks/refresh.py [hunk 7]
```diff
@@ -20,9 +22,10 @@ def refresh():
...
         except Exception as error:
             manage.logger.exception(error)
-    try:
-        mail.send_mail(buffer)
-    except Exception as error:
-        logger.exception(error)
+            slack_message = '*ERROR* nightly update failed for {0}. Check logs.'.format(env.get_credential(NEW_RELIC_APP_NAME))
+            utils.post_to_slack(slack_message, '#bots')
```
Comment (ccostino) [inline webservices/tasks/refresh.py]: Same notes as above for the `NEW_RELIC_APP_NAME` and Slack channel name.

File: webservices/utils.py [hunk 8]
```diff
@@ -335,3 +340,15 @@ def create_eregs_link(part, section):
     if section:
         url_part_section += '-' + section
     return '/regulations/{}/CURRENT'.format(url_part_section)
+
+def post_to_slack(message, channel):
+    response = requests.post(
+        env.get_credential('SLACK_HOOK'),
```
Comment (ccostino) [inline webservices/utils.py]: Does the `SLACK_HOOK` environment variable need to be managed outside in the credential service?

File: webservices/utils.py [hunk 9]
```diff
@@ -335,3 +340,15 @@ def create_eregs_link(part, section):
     if section:
         url_part_section += '-' + section
     return '/regulations/{}/CURRENT'.format(url_part_section)
+
+def post_to_slack(message, channel):
```
Comment (ccostino) [inline webservices/utils.py]: As mentioned before, what if we set an environment variable for the default Slack channel to post to, but keep the parameter as you have to so we can override it.

File: webservices/utils.py [hunk 10]
```diff
@@ -335,3 +340,15 @@ def create_eregs_link(part, section):
...
+    response = requests.post(
+        env.get_credential('SLACK_HOOK'),
+        data=json.dumps({
+            'text': message, 'channel': channel, 'link_names': 1,
+            'username': 'Ms. Robot', 'icon_emoji': ':robot_face:',
```
Comment (ccostino) [inline webservices/utils.py]: ❤️ 😃

File: requirements.txt (hunk 3 above)
Comment (LindsayYoung) [reply requirements.txt]: yea it was too easy to justify importing a dependency

File: webservices/common/views.py (hunk 4 above)
Comment (LindsayYoung) [reply webservices/common/views.py]: Nice catch!

File: webservices/tasks/refresh.py (hunk 6 above)
Comment (LindsayYoung) [reply webservices/tasks/refresh.py]: I was a way to get text as to which environment it was running on without adding more variables

File: webservices/utils.py (hunk 9 above)
Comment (LindsayYoung) [reply webservices/utils.py]: Documenting here: We talked it over and it is easier to have a default that we don't have to configure across all environments, we could do an optional value you pass in the future.

File: webservices/utils.py (hunk 8 above)
Comment (LindsayYoung) [reply webservices/utils.py]: I think we need to add this to circle- I will do that now

Comment (LindsayYoung) [reply webservices/utils.py]: done, now we can have things slack us in circle if we want it to

File: webservices/config.py (hunk 5 above)
Comment (LindsayYoung) [reply webservices/config.py]: I thought about it but, I would not want not having the slack token to cause the build to fail. I assume we will notice that it is broken.

File: manage.py (hunk 2 above)
Comment (ccostino) [reply manage.py]: We opted not to after discussing this more - if we find that this'd be useful we can always do it in the future!

**Pull request merged**